### PR TITLE
Protect ProposerPolicy Registry using Mutex

### PR DESCRIPTION
**Pull request by jbhurat**
`RegisterValidatorSet()` might throw a nil pointer, if the `validatorSet` registry is cleared in `ClearRegistry()` by another goroutine. To fix this access to `registry` needs to be protected.

### CODE REVIEW

File: consensus/istanbul/backend/engine_test.go [hunk 1]
```diff
@@ -169,14 +169,14 @@ func TestSealStopChannel(t *testing.T) {
...
 	}
-	go eventLoop()
 	resultCh := make(chan *types.Block, 10)
 	go func() {
 		err := engine.Seal(chain, block, resultCh, stop)
 		if err != nil {
 			t.Errorf("error mismatch: have %v, want nil", err)
 		}
 	}()
+	go eventLoop()
```
Comment (ricardolyn) [inline consensus/istanbul/backend/engine_test.go:179]: this change seems unrelated to title. what was it about?

**Review by ricardolyn — APPROVED** [review]
LGTM

File: consensus/istanbul/backend/engine_test.go (hunk 1 above)
Comment (jbhurat) [reply consensus/istanbul/backend/engine_test.go:179]: Yes, the change is unrelated to the PR. When I created the PR, this test failed. On looking closely, the test was failing like 1 in 250 - 300 runs. Moving the call to `eventLoop()` ensures that `stop` channel is not triggered before `Seal()` is called
